js/src/components/tree-select-control/index.js

```diff
@@ -169,6 +177,10 @@ const TreeSelectControl = ( {
 	const handleParentChange = ( checked, option ) => {
 		const newValue = [ ...value ];
 
+		if ( checked && ! nodesExpanded.includes( option.value ) ) {
+			setNodesExpanded( [ ...nodesExpanded, option.value ] );
+		}
```

Comment:
❔  I'm not quite sure it only needs to expand the children of the first layer or children of all layers in the sub-tree. Maybe needs to confirm with the design team.

Reply:
I will